logstash-core/src/main/java/org/logstash/plugins/ConfigVariableExpander.java

```diff
@@ -87,7 +90,11 @@ public Object expand(Object value) {
             if (secretStore != null) {
                 byte[] ssValue = secretStore.retrieveSecret(new SecretIdentifier(variableName));
                 if (ssValue != null) {
-                    return new String(ssValue, StandardCharsets.UTF_8);
+                    if (keepSecrets) {
+                        return new Password(new String(ssValue, StandardCharsets.UTF_8));
```

Comment:
Password doesn't seem like the right concept to use here. While there is a similar behaviour we want from the Password class, this is a broader concept, a Secret.
Using Password here can cause some confusion when printing the logs, where a pipeline like

```
input { stdin {} } output { if \"\${my_secret}\" == [message] { stdout { codec => rubydebug } } }
```
Will cause debug logs like:
```
[2022-04-20T10:40:22,268][DEBUG][org.logstash.config.ir.CompiledPipeline][main] Compiled conditional
 [if (<password>==event.getField('[message]'))] 
 into 
 org.logstash.config.ir.compiler.ComputeStepSyntaxElement@9fb449bc
```
And it'd be better it was like:
```
[2022-04-20T10:40:22,268][DEBUG][org.logstash.config.ir.CompiledPipeline][main] Compiled conditional
 [if (<secret>==event.getField('[message]'))] 
 into 
 org.logstash.config.ir.compiler.ComputeStepSyntaxElement@9fb449bc
```
Or even better:
```
[2022-04-20T10:40:22,268][DEBUG][org.logstash.config.ir.CompiledPipeline][main] Compiled conditional
 [if (${my_secret}==event.getField('[message]'))] 
 into 
 org.logstash.config.ir.compiler.ComputeStepSyntaxElement@9fb449bc
```
By having the Secret class be constructed with the key as well as the value, and then `toString() { return "${" + key + "}" }`

This would facilitate debugging if necessary.

Reply:
I like it! :-D nice suggestion

Reply:
Suggestion implemented, so ready for a review :-)
- commit [8784d31](https://github.com/elastic/logstash/pull/13997/commits/8784d31216cd9bb350c1abae877b2a37f83b91db) introduce the `SecretVariable` class
- commit [79759c6](https://github.com/elastic/logstash/pull/13997/commits/79759c6aa04b34cc1bb3363d45e6cdc0d135229a) is a refactoring to streamline a method flow